File: nexus-aura-backend/src/main/java/com/nexusaura/nexus_aura_backend/model/Post.java
package com.nexusaura.nexus_aura_backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "Post")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer postID;

    @ManyToOne
    @JoinColumn(name = "userID", nullable = false)
    private UserEntity user;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(length = 20)
    private String mediaType;

    @Column(length = 255)
    private String mediaURL;

    @Column(nullable = false)
    private LocalDateTime timestamp;
}

File: nexus-aura-backend/src/main/kotlin/com/nexus/aura/backend/nexus_aura_backend/entity/Post.kt
package com.nexus.aura.backend.nexus_aura_backend.entity

import java.util.*
import jakarta.persistence.*

@Entity
@Table(name = "posts")
data class Post(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: UUID = UUID.randomUUID(),

    @Column(nullable = false)
    var content: String,

    @Column(nullable = false)
    val createdAt: Date = Date(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    var user: User,

    @Column(nullable = false)
    var likes: Int = 0,
)
